src/main/java/seedu/taassist/storage/JsonAdaptedStudent.java

```diff
@@ -29,22 +27,22 @@ class JsonAdaptedStudent {
     private final String email;
     private final String address;
 
-    @JsonProperty("classes")
-    private final List<String> moduleClasses = new ArrayList<>();
+    @JsonProperty("moduleData")
+    private final List<JsonAdaptedStudentModuleData> moduleData = new ArrayList<>();
 
     /**
      * Constructs a {@code JsonAdaptedStudent} with the given student details.
      */
     @JsonCreator
     public JsonAdaptedStudent(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-            @JsonProperty("email") String email, @JsonProperty("address") String address,
-            @JsonProperty("classes") List<String> moduleClasses) {
+                              @JsonProperty("email") String email, @JsonProperty("address") String address,
+                              @JsonProperty("moduleData") List<JsonAdaptedStudentModuleData> moduleData) {
```

Comment:
Does this pass the checkstyle? I think the checkstyle sometimes complains when there are extra spaces in front of code even though we are trying to align the arguments. Still, it might be nice to stick to default argument spacing to be consistent with other multiline method headers.